src/handlers/backport.rs

```diff
@@ -0,0 +1,227 @@
+use std::collections::HashMap;
+use std::sync::LazyLock;
+
+use crate::config::BackportTeamConfig;
+use crate::github::{IssuesAction, IssuesEvent, Label};
+use crate::handlers::Context;
+use regex::Regex;
+use tracing as log;
+
+// see https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue
+static CLOSES_ISSUE_REGEXP: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new("(?i)(close[sd]*|fix([e]*[sd]*)?|resolve[sd]*) #(\\d+)").unwrap());
+
+const BACKPORT_LABELS: [&str; 4] = [
+    "beta-nominated",
+    "beta-accepted",
+    "stable-nominated",
+    "stable-accepted",
+];
+
+#[derive(Default)]
+pub(crate) struct BackportInput {
+    // Issue(s) fixed by this PR
+    ids: Vec<u64>,
+    // Labels profile, compound value of (needs_label -> add_labels)
+    profile_labels: HashMap<String, Vec<String>>,
+}
+
+pub(super) async fn parse_input(
+    _ctx: &Context,
+    event: &IssuesEvent,
+    config: Option<&BackportTeamConfig>,
+) -> Result<Option<BackportInput>, String> {
+    let config = match config {
+        Some(config) => config,
+        None => return Ok(None),
+    };
+
+    if !matches!(event.action, IssuesAction::Opened) && !event.issue.is_pr() {
```

Comment:
Contributors often add labels after opening the PR, we should probably also listen to "Edited" events, but that becomes expansive.

What about only triggering the logic when the PR is merged? That way we reduce the false-positives and false-negatives.

Reply:
Hm ... in general I think that having more time to evaluate backporting a patch is better, therefore nominating it while the PR is in progress is a good thing. We live in different time zones and sometimes the "voting" time window is short due to various factors.

Reply:
Fine by me.

We should still however look at the "Edited" events, can be done later of-course.